### PR TITLE
removes `gogoproto.equal = false` directive for protos

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

this directive defaults to `false`, so we don't actually need to set it to false.

closes: #3871 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/client/v1/client.proto [hunk 1]
```diff
@@ -57,9 +57,9 @@ message ClientUpdateProposal {
 // UpgradeProposal is a gov Content type for initiating an IBC breaking
 // upgrade.
 message UpgradeProposal {
-  option (gogoproto.goproto_getters)         = false;
-  option (gogoproto.goproto_stringer)        = false;
-  option (gogoproto.equal)                   = true;
```
Comment (damiannolan) [inline proto/ibc/core/client/v1/client.proto:62]: this one is true, is it required or fine to remove?

**Review by damiannolan — APPROVED** [review]

File: proto/ibc/core/client/v1/client.proto (hunk 1 above)
Comment (charleenfei) [reply proto/ibc/core/client/v1/client.proto:62]: agh. i didnt intend to remove this one actually. i'll add it back

Comment (chatton) [reply proto/ibc/core/client/v1/client.proto:62]: I don't see any usages of its equal function, but also curious!

**Review by chatton — APPROVED** [review]
LGTM when all the tests are 🟢

File: proto/ibc/core/client/v1/client.proto (hunk 1 above)
Comment (damiannolan) [reply proto/ibc/core/client/v1/client.proto:62]: I don't see why we would need to check for equality on upgrade proposals, it makes sense for `Coin` tho

**Pull request merged**